### PR TITLE
feat(modal): sync with @carbon/react v11

### PR DESCRIPTION
**Pull request by IgnacioBecerra**
### Related Ticket(s)
[#10105](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/10105)

### Description
This PR syncs the Modal component to the Carbon React v11 version.

### Changelog

**New**

- `alert`, `danger`, `hasScrollingContent`, `fullWidth`, and more new props
- new `cds-modal-body-content` to target correct styles instead of `<p>` due to styling issues with shadow DOM

**Changed**

- cleaned styles

### CODE REVIEW

**Review by RichKummer — APPROVED** [review]
LGTM @IgnacioBecerra !

File: packages/carbon-web-components/src/components/modal/defs.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export enum MODAL_SIZE {
   /**
    * Regular size.
```
Comment (ariellalgilmore) [inline packages/carbon-web-components/src/components/modal/defs.ts:25]:
```suggestion
   * Medium size.
```

File: packages/carbon-web-components/src/components/modal/modal-close-button.ts [hunk 2]
```diff
@@ -23,21 +23,21 @@ import styles from './modal.scss';
  * @csspart close-icon The close icon.
  */
 @customElement(`${prefix}-modal-close-button`)
-class BXModalCloseButton extends FocusMixin(LitElement) {
+class CDSModalCloseButton extends FocusMixin(LitElement) {
   /**
-   * The assistive text for the button.
+   * Specify an label for the close button of the modal; defaults to close
```
Comment (ariellalgilmore) [inline packages/carbon-web-components/src/components/modal/modal-close-button.ts:28]:
```suggestion
   * Specify a label for the close button of the modal; defaults to close
```

**Review by ariellalgilmore — COMMENTED** [review]
LGTM!!!

 the only thing i noticed that seems like a storybook specific issue is this kind of weird jumping when closing the modal
Apr-27-2023 08-43-29

**Comment by IgnacioBecerra** [comment]
@ariellalgilmore addressed the comments!

**Review by ariellalgilmore — APPROVED** [review]
Looks awesome!!

**Pull request merged**